packages/peregrine/lib/talons/CheckoutPage/PaymentInformation/usePaymentInformation.js

```diff
@@ -115,6 +136,27 @@ export const usePaymentInformation = props => {
         onCompleted: onPaymentDetailsQueryCompleted
     });
 
+    const handleExiredPaymentError = useCallback(() => {
+        clearPaymentDetails({ variables: { cartId } });
+        resetReviewOrderButtonClicked();
+        setHasData(false);
+        onError();
+    }, [resetReviewOrderButtonClicked, onError, clearPaymentDetails, cartId]);
+
+    /**
+     * Effects
+     */
+
+    useEffect(() => {
+        if (
+            checkoutError &&
+            checkoutError instanceof CheckoutError &&
+            checkoutError.hasPaymentExpired()
+        ) {
+            handleExiredPaymentError();
```

Comment:
Since `handleEx(p)iredPaymentError` is only called here I would just call the `clearPaymentDetails` and `setHasData` functions directly here.

Reply:
It is just a choice of style. I would prefer to leave the 4 function calls in their own function. So when someone reads the if condition, they will know that the expired token will be handled. They don't need to know how, unless if they want to, they can check the function itself.